src/main/java/org/truffleruby/language/globals/GlobalVariableStorage.java

```diff
@@ -18,6 +18,8 @@
 
 public class GlobalVariableStorage {
 
+    private static final Object UNSET_VALUE = new Object();
```

Comment:
Could this be `NotProvided.INSTANCE` to make it easier to debug?

Reply:
Or otherwise give it a class so you don't just see `Object` when you're debugging in the future.